frontend/src/packages/editor/components/buttons/EditorCodeButton.tsx

```diff
@@ -0,0 +1,20 @@
+import React, { FC } from "react";
+import { useSlate } from "slate-react";
+import CodeButton from "src/cse-ui-kit/small_buttons/CodeButton";
+import { toggleMark } from "./buttonHelpers";
+
+const EditorCodeButton: FC = () => {
+  const editor = useSlate();
+  return (
+    <CodeButton
+      size={30}
+      onMouseDown={(event) => {
+        event.preventDefault();
+        // TODO switch mark to code (not sure if this has to be defined)
```

Comment:
i believe we toggle mark as it helps present the array of the text in database with a mark that the following contents is a inline code block. We then use this mark in the homepage to display that text as code. Feel free to ask me about this in more depth if this doesn't make any sense :D
e.g. `[{"align":"left","quote":true,"underline":true,"code":true,"text":"sndfgjsbdfjg"}]}]`

Reply:
once it is clear feel free to remove the comment and merge into main :)

Reply:
all good! just forgot to delete the comment